### ISSUE TITLE
2.5.3 is visible select option considered part of the accessible name here?

### ISSUE THREAD

**Issue by detlevhfischer** [w3c/wcag#2484]
The situation is simple but the [2.5.3 understanding text](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html) is not explicit regarding the assessment of visible select default options.

Is this a FAIL of 2.5.3?
A `select` with:

- no visible (and prgrammatically linked) `label` 
- a default visible `option` "Addresses used"
- a `title` with the value "previous destinations"

Screenreaders I tested render both the `title` AND (after a short gap) the default `option`. Still, the question is whether in a speech input scenario, the assistive technology would support something like "click _Addresses used_". And beyond current AT support, the general question whether a strict interpratation of "accessible name" for a `select` would disregard altogether the default option (and only accept a programmatically linked `label`, an `aria-label`, text referenced by `aria-labelledby`, or `title`).

**Comment by detlevhfischer** [comment]
Thanks @scottaohara  - that is helpful and confirms my suspicion.
@bruce-usab I believe that `title` **can** be used as accname of a `select` - the example mentioned just fails 2.5.3 (and may in addition fail 3.3.2)

**Comment by detlevhfischer** [comment]
@scottaohara If the default option "Addresses used" can be considered visually sufficient and any other option chosen implicity shows it is an address (which in the context of supplying origin or destination information for a journey seems pretty clear to me) one could argue that there is always a marginally sufficient visual label: either the default, or the chosen address after interaction with the select. So this is a bit different from the placeholder case where  the visual label disappears with any input. When the author ensures that accname = default option, all but this default would then fail 2.5.3 after any selection?

**Comment by scottaohara** [comment]
@detlevhfischer - i think you misunderstood my comment? I wasn't implying anything about 3.3.2 having anything to do with being programmatically available, nor did i mean 'placeholder' to correlate to the `placeholder` attribute or the floating label technique.

>any other select option implicity shows it is an address

sure, but does the value represent an "address", or a billing address, shipping, delivery, destination, location?  Depending on other information in the form, this could be unclear and require someone to have to re-open the listbox popup to see the initial 'placeholder' option (again, not a label and not a name).

If using the `title` attribute, someone on a touch device would never get to see the actual name/label of the element.

**Comment by detlevhfischer** [comment]
@scottaohara no, I think I got you comment, and sorry, I I did not mean to suggest that you think 3.2.2 is related to programmatic determination - I just wanted to pad out my line of argument that if, in some context, you consider both default optic and any selection sufficient to meet 3.3.2 there would still be an issue with 2.5.3. I have not (could not render the full example - in the case it is travel origin and travel destination an the select option holds street, place and post  code.

> If using the title attribute, someone on a touch device would never get to see the actual name/label of the element.

true, but `title` would (and should, to meet 2.5.3) match the visible option here - but this would fall apart once a selection is made and the default option is replaced. I would assume label in name would be required for **all** states and for this reason, using option as a visible label would never work across all situations. That is why I would still be inclined to fail this case under 2.5.3 rather than 3.3.2 - but I see that a different argument can be made.

**Comment by fstrr** [comment]
This issue is labelled as a discussion, so we’re moving this to Discussions. There doesn’t seem to be an update to make to the documentation, but if that changes, we can move it back to the issues list.
